pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -1757,60 +1702,23 @@ func (s *AWSCloud) updateInstanceSecurityGroupsForLoadBalancer(lb *elb.LoadBalan
 		return fmt.Errorf("Could not determine security group for load balancer: %s", orEmpty(lb.LoadBalancerName))
 	}
 
-	// Get the actual list of groups that allow ingress from the load-balancer
+	// Gather all security groups
 	describeRequest := &ec2.DescribeSecurityGroupsInput{}
-	filters := []*ec2.Filter{}
-	filters = append(filters, newEc2Filter("ip-permission.group-id", loadBalancerSecurityGroupId))
-	describeRequest.Filters = s.addFilters(filters)
```

Comment:
agreeing with @justinsb on this one. @chrisleck could you write a script to clean up the SGs vs permanently implementing the cleaning up in k8s? i'd really like to see this merged.